lib/src/io.dart

```diff
@@ -1036,46 +1036,33 @@ ByteStream createTarGz(List contents, {String baseDir}) {
         "--directory",
         baseDir
       ];
+      var stdin = "";
 
       if (Platform.isLinux) {
         // GNU tar flags.
         // https://www.gnu.org/software/tar/manual/html_section/tar_33.html
 
         args.addAll(["--files-from", "/dev/stdin"]);
+        stdin = contents.join("\n");
 
         // The ustar format doesn't support large UIDs. We don't care about
         // preserving ownership anyway, so we just set them to "pub".
         args.addAll(["--owner=pub", "--group=pub"]);
-
-        var process = await startProcess("tar", args);
-        process.stdin.add(UTF8.encode(contents.join("\n")));
-        process.stdin.close();
-        return process.stdout;
       } else {
         // OSX tar flags, applicable since at least OSX 10.9 (bsdtar 2.8.3)
         // https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/tar.1.html
 
-        // Create the file containing the list of files to compress.
-        var tempDir = createSystemTempDir();
-
-        try {
-          // Create the file containing the list of files to compress.
-          var contentsPath = path.join(tempDir, "files.mtree");
-
-          // The ustar format doesn't support large UIDs, so we set
-          // UIDs and GIDs to 0 via an mtree input file.
-          var uidGid = " uid=0 gid=0 type=file";
-          writeTextFile(contentsPath, "#mtree\n" + contents.join("$uidGid\n") + uidGid + "\n");
-          args.add("@${contentsPath}");
-
-          return (await startProcess("tar", args, workingDir: baseDir))
-              .stdout
-              .transform(onDoneTransformer(() => deleteEntry(tempDir)));
-        } catch (_) {
-          deleteEntry(tempDir);
-          rethrow;
-        }
+        // The ustar format doesn't support large UIDs, so we set
+        // UIDs and GIDs to 0 via an mtree input file.
+        args.add("@/dev/stdin");
+        var uidGid = " uid=0 gid=0 type=file\n";
```

Comment:
From [this page](https://www.freebsd.org/cgi/man.cgi?mtree(8)), it looks like the mtree format supports symbolic names for the user and group rather than numeric values. If we can set the symbolic names to "pub", then we'll be generating the same files on OS X as we do on Linux, which is nice. It also means manually-extracted package archives won't have root permissions, which is a bonus.